File: arrays_and_hashing/problems/349_intersection_of_two_arrays.py
# Given two integer arrays nums1 and nums2, return an array of their intersection. Each element in the result must be unique and you may return the result in any order.

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        N1 = set(nums1)
        N2 = set(nums2)
        ans = []
        for n in N1:
            if n in N2:
                ans.append(n)
        return ans
        
        

File: typescript/problems/349_intersection_of_two_arrays.ts
/**
Given two integer arrays nums1 and nums2, return an array of their intersection. Each element in the result must be unique and you may return the result in any order.
 */

function intersection(nums1: number[], nums2: number[]): number[] {
    const N1: Set<number> = new Set(nums1);
    const N2: Set<number> = new Set(nums2);
    let ans: number[] = [];
    for (const n of N1) {
        if (N2.has(n)) {
            ans.push(n)
        }
    }
    return ans
};